app/src/main/java/com/example/smkitdemoapp/viewModels/ActivityViewModel.kt

```diff
@@ -0,0 +1,124 @@
+package com.example.smkitdemoapp.viewModels
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.PointF
+import android.util.Log
+import android.view.View
+import android.widget.Toast
+import androidx.camera.core.Preview
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.example.smkitdemoapp.models.DemoExercise
+import com.example.smkitdemoapp.models.ExerciseState
+import com.example.smkitdemoapp.models.State
+import com.sency.smkit.SMKit
+import com.sency.smkit.listener.ConfigurationResult
+import com.sency.smkit.listener.DetectionDataListener
+import com.sency.smkit.listener.SMKitSessionListener
+import com.sency.smkit.model.FrameInfo
+import com.sency.smkit.model.SMKitJoint
+import com.sency.smkit.model.SMKitMovementData
+
+class ActivityViewModel: ViewModel() {
+
+    private val exerciseList = mutableListOf<DemoExercise>()
+    private var smKit: SMKit? = null
+
+    private val _state = MutableLiveData(initState())
+    val state: LiveData<State> get() = _state
+
+    private val _exerciseState = MutableLiveData(initExerciseState())
+    val exerciseState: LiveData<ExerciseState> get() = _exerciseState
+
+    private val _repCounter = MutableLiveData(0)
+    val repCounter: LiveData<Int> get() = _repCounter
+
+    fun addExercise(exercise: DemoExercise) {
+        exerciseList.add(exercise)
+    }
+
+    private val configureListener = object: ConfigurationResult {
+        override fun onFailure() {
+            val state = _state.value
+            _state.postValue(state?.copy(configure = false))
+        }
+
+        override fun onSuccess() {
+            val state = _state.value
+            _state.postValue(state?.copy(configure = true))
+            val exerciseState = _exerciseState.value
+            val name = exerciseList[0].name
+            _exerciseState.postValue(exerciseState?.copy(exerciseIndex = 0, exerciseName = name))
+        }
+    }
+
+    fun configure(context: Context) {
+        smKit = SMKit.Builder(context).authKey("public_live_#gdz3t)mW#\$39Crs").isUI(false).build()
```

Comment:
@SencyCaspit since this is a real key we need to create another branch and not commit this key at all to keep it out of commit history